cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -123,6 +135,21 @@ public AppFabricServer(CConfiguration configuration, DiscoveryService discoveryS
     this.pluginService = pluginService;
     this.privilegesFetcherProxyService = privilegesFetcherProxyService;
     this.routeStore = routeStore;
+    this.sslEnabled = cConf.getBoolean(Constants.Security.Ssl.SSL_INTERNAL_ENABLED);
+    if (sslEnabled) {
+      this.serverPort = cConf.getInt(Constants.AppFabric.SERVER_SSL_PORT);
+      KeyStore ks = GeneratedCertKeyStores.getSSLKeyStore(sConf);
```

Comment:
Will this generate a new cert? If it does, it can potentially take quite a while. Better do it in `start` rather than in constructor.